### PR TITLE
[PyAPI][Opset14] Python API ConvertPromoteTypes-14

### PR DESCRIPTION
**Pull request by mmikolajcz**
### Details:
 - *Python API for ConvertPromoteTypes-14*
 - *...*

### Tickets:
 - *129205*

### CODE REVIEW

**Comment by mlukasze** [comment]
@openvinotoolkit/openvino-ie-python-api-maintainers please review

File: src/bindings/python/src/openvino/runtime/opset14/ops.py [hunk 1]
```diff
@@ -4,30 +4,48 @@
...
 
 
 # -------------------------------------------- ops ------------------------------------------------
+@nameable_op
+def convert_promote_types(
+    left_node: NodeInput,
+    right_node: NodeInput,
+    promote_unsafe: bool = False,
+    pytorch_scalar_promotion: bool = False,
+    u64_integer_promotion_target: str = "f32",
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/opset14/ops.py:23]:
```suggestion
    u64_integer_promotion_target: Union[str, ov.Type] = "f32",
```
Would it be possible to pass OV types as well? (*)

File: src/bindings/python/src/openvino/runtime/opset14/ops.py [hunk 2]
```diff
@@ -4,30 +4,48 @@
...
+    :param u64_integer_promotion_target: Element type attribute to select promotion result when inputs are u64 and signed integer.
+
+    :return: The new node performing ConvertPromoteTypes operation.
+    """
+    inputs = as_nodes(left_node, right_node)
+
+    attributes = {
+        "promote_unsafe": promote_unsafe,
+        "pytorch_scalar_promotion": pytorch_scalar_promotion,
+        "u64_integer_promotion_target": u64_integer_promotion_target,
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/opset14/ops.py:40]:
```suggestion
        "u64_integer_promotion_target": u64_integer_promotion_target.to_string() if isinstance(u64_integer_promotion_target, ov.Type) else u64_integer_promotion_target,
```
(*) just an idea how to get it working if ov.Type is not accepted by default.

**Review by jiwaszki — APPROVED** [review]
Overall OK, just small improvement proposal.

File: src/bindings/python/src/openvino/runtime/opset14/ops.py (hunk 1 above)
Comment (mmikolajcz) [reply src/bindings/python/src/openvino/runtime/opset14/ops.py]: Thanks, done, ov Type can be passed. Modified tests to use both str and Type

File: src/bindings/python/src/openvino/runtime/opset14/ops.py (hunk 2 above)
Comment (mmikolajcz) [reply src/bindings/python/src/openvino/runtime/opset14/ops.py:41]: This wasn't necessary, ov Type seem to be converted by default. Modified test cases to use both Type and str

File: src/bindings/python/tests/test_graph/test_convert_promote_types.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        (([], np.uint8), ([4, 4], np.float16), Type.f16),
+    ],
+)
+def test_convert_promote_types_default_attrs(lhs, rhs, expected_output_type):
+    lhs_param = ops.parameter(*lhs)
+    rhs_param = ops.parameter(*rhs)
+    op = ops.convert_promote_types(lhs_param, rhs_param)
+    attrs = op.get_attributes()
+    assert attrs.get("promote_unsafe") is False
+    assert attrs.get("pytorch_scalar_promotion") is False
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_convert_promote_types.py:82]:
```suggestion
    assert not attrs.get("promote_unsafe")
    assert not attrs.get("pytorch_scalar_promotion")
```

**Review by akuporos — APPROVED** [review]

**Pull request merged**